src/test/scala/cromwell/binding/ScatterSpec.scala

```diff
@@ -0,0 +1,70 @@
+package cromwell.binding
+
+import cromwell.parser.BackendType
+import cromwell.util.SampleWdl
+import org.scalatest.{FlatSpec, Matchers}
+
+class ScatterSpec extends FlatSpec with Matchers {
+  val namespace = NamespaceWithWorkflow.load(SampleWdl.ScatterWdl.wdlSource(), BackendType.LOCAL)
+
+  it should "Have four 'children' objects" in {
+    namespace.workflow.children.size shouldEqual 4
+  }
+
+  it should "Have two 'direct Call descendents' objects" in {
+    Scope.collectCalls(namespace.workflow.children).size shouldEqual 2
+  }
+
+  it should "Have two 'direct Scatter descendents' objects" in {
+    Scope.collectScatters(namespace.workflow.children).size shouldEqual 2
+  }
+
+  it should "Have eight 'Call' objects" in {
+    namespace.workflow.calls.size shouldEqual 8
+  }
+
+  it should "Have four 'Scatter' objects" in {
+    namespace.workflow.scatters.size shouldEqual 4
+  }
+
+  it should "Have 'Scatter' objects indexed properly" in {
+    namespace.workflow.scatters(0).index shouldEqual 0
+    namespace.workflow.scatters(1).index shouldEqual 3
+    namespace.workflow.scatters(2).index shouldEqual 1
+    namespace.workflow.scatters(3).index shouldEqual 2
```

Comment:
These tests aren't "DRY" enough [IMHO](http://www.urbandictionary.com/define.php?term=IMHO). Perhaps something like:

```scala
it should "Have 'Scatter' objects indexed properly" in {
  namespace.workflow.scatters.map(_.index) should contain inOrderOnly(0, 3, 1, 2)
}

it should "Not appear in Calls FQNs" in {
  val calls: Seq[Call] = namespace.workflow.calls
  // NOTE: This still halts on the first error, instead of returning all the undefined entries.
  ('A' to 'H') foreach { x => calls.find(_.fullyQualifiedName == s"w.$x") shouldBe defined }
}

// ... and similar for the others below ...
```

I don't believe these tests are candidates yet to go past basic [matchers](http://www.scalatest.org/user_guide/using_matchers). However, as we get even closer to implementing a full scatter stack, we may consider separating out [property](http://www.scalatest.org/user_guide/property_based_testing) matching with [tables](http://www.scalatest.org/user_guide/table_driven_property_checks) in the future. That would allow a 4x2 table of visually laid out values, as the above four lines, but separated from one line of actual code.